Some changes to the functionality

mme89 opened the pull request:
- Dark UI variables and panels
- Channel grid/tag filtering LiveView
- Graphs page: top channels/users, watch time, stream type distribution
- History filters (search, date range, status)
- Server info + active stream enhancements
- API client: channel/tag helpers, server info"

No expert here, when it comes to elixir. but its running in my dev environment.
you can argue about the design changes I made. But added a few new API calls and stuff.

mme89 committed:
Some changes to the functionality
- Dark UI variables and panels
- Channel grid/tag filtering LiveView
- Graphs page: top channels/users, watch time, stream type distribution
- History filters (search, date range, status)
- Server info + active stream enhancements
- API client: channel/tag helpers, server info"

jbonet commented:
Hello,

Thanks for the work you've put into this.

I have some questions about it. 

- Can you provide an screenshot of how it looks like? In my dev environment (which may be broken) it's difficult to see some parts of the dashboard.
- How could you get the dev env to work? I've had lots of trouble to recreate it, to the point where the docker image wouldn't build in the current state.

<img width="2011" height="705" alt="image" src="https://github.com/user-attachments/assets/3c5fe3cc-ce6f-498b-9854-ff5cbbbf6c78" />

See the current contrast, makes it difficult to see.

- I love the new filtering system. I have two questions about it:
  - It takes up a lot of space, could you make it collapsible, at least on mobile?
  - The user field, could be a precomputed select, from the available users in the server/database instead of free form text.


PS: This project is running elixir 1.14.1, which dates from Sept 2022. I will try to update it to a more recent version so it's easier to work with.

mme89 commented:
I started few month ago playing with this, I can't tell you why I did it, but I changed the dev Dockerfile to debian.
```
ARG ELIXIR_VERSION=1.14.1
ARG OTP_VERSION=25.1.1
ARG DEBIAN_VERSION=bullseye-20220801

FROM hexpm/elixir:${ELIXIR_VERSION}-erlang-${OTP_VERSION}-debian-${DEBIAN_VERSION}

# Set Locale
ENV LANG=C.UTF-8
ENV LC_ALL=C.UTF-8
ENV PHOENIX_VERSION=1.6.13

# Add inotify-tools and essentials
RUN apt-get update -y && apt-get install -y --no-install-recommends \
        inotify-tools wget ca-certificates bash libstdc++6 \
    && apt-get clean && rm -rf /var/lib/apt/lists/*

# Create required dirs
RUN mkdir /app

# Install hex and rebar
RUN mix local.hex --force
RUN mix local.rebar --force

# Install phoenix
RUN mix archive.install hex phx_new $PHOENIX_VERSION --force

# Set /app as workdir
WORKDIR /app
```

also made some changes to the entrypoint.sh.

```
#!/usr/bin/env bash

mix deps.get
mix ecto.migrate

mix sass.install --if-missing >/dev/null 2>&1 || true
mix tailwind.install --if-missing >/dev/null 2>&1 || true
mix esbuild.install --if-missing >/dev/null 2>&1 || true

exec "$@"
```

screenshots coming later

mme89 committed:
add user filtering dropdown + adjust active stream
- Add user dropdown to history page
- Display streaming profile in active streams panel

mme89 commented:
to be honest, the design is all over the place. but I am no designer or ui designer.
it definitely needs some work, to get it more clean and also work better with mobile.

<img width="2002" height="872" alt="Bildschirmfoto 2025-09-14 um 14 54 31" src="https://github.com/user-attachments/assets/a4aaa6b6-6980-476c-a327-9ea3a8c03c71" />
<img width="2003" height="1067" alt="Bildschirmfoto 2025-09-14 um 14 54 48" src="https://github.com/user-attachments/assets/48064412-0cb1-4807-859b-65231fdef035" />
<img width="2003" height="756" alt="Bildschirmfoto 2025-09-14 um 14 55 01" src="https://github.com/user-attachments/assets/f33db2ab-42ad-4468-a72b-3a34b5169986" />
<img width="2000" height="1207" alt="Bildschirmfoto 2025-09-14 um 14 55 20" src="https://github.com/user-attachments/assets/e64f4b2d-f30b-4cc9-968d-6496565d8882" />
